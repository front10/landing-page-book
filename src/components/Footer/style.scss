:root {
    --Footer-color: #cccccc;
    --Footer-backgroundColor: #383d45;
    --Footer__Sections__Link--hover-color: #fed136;
    --Footer__Sections__Header-fontSize: 1.5rem;
    --Footer__Sections__Link-fontSize: 0.8rem;
    --Footer__Copyright-fontSize: 0.8rem;
}

.Footer {
    padding: 2.5rem 0;
    color: var(--Footer-color);
    text-align: center;
    background-color: var(--Footer-backgroundColor);
    .Footer__Copyright {
        font-size: var(--Footer__Copyright-fontSize);
    }
    .Footer__Sections__Header {
        font-size: var(--Footer__Sections__Header-fontSize);
    }
    a {
        color: var(--Footer-color);
        font-size: var(--Footer__Sections__Link-fontSize);
    }
    a:hover {
        color: var(--Footer__Sections__Link--hover-color);
    }
}