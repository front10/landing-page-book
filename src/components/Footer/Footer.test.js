import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Footer from './Footer';

configure({ adapter: new Adapter() });

describe('Footer', () => {
  let props;
  let mounted;
  const footer = () => {
    if (!mounted) {
      mounted = mount(<Footer {...props} />);
    }
    return mounted;
  };
  describe('footer', () => {
    beforeEach(() => {
      props = {
        sections: [
          {
            name: 'Features',
            sections: [
              {
                name: 'Cool stuff',
                url: 'https://front10.com'
              },
              {
                name: 'Random feature',
                url: 'https://front10.com'
              },
              {
                name: 'Team feature',
                url: 'https://front10.com'
              },
              {
                name: 'Stuff for developers',
                url: 'https://front10.com'
              },
              {
                name: 'Another one',
                url: 'https://front10.com'
              },
              {
                name: 'Last time',
                url: 'https://front10.com'
              }
            ]
          }]
      };
      mounted = undefined;
    });

    it('should render', () => {
      expect(mount(<footer {...props} />)).toMatchSnapshot();
    });

    it('always renders a div', () => {
      const divs = footer().find('footer');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});
