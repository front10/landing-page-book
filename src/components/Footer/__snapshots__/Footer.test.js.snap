// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer footer should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <footer
    sections={
      Array [
        Object {
          "name": "Features",
          "sections": Array [
            Object {
              "name": "Cool stuff",
              "url": "https://front10.com",
            },
            Object {
              "name": "Random feature",
              "url": "https://front10.com",
            },
            Object {
              "name": "Team feature",
              "url": "https://front10.com",
            },
            Object {
              "name": "Stuff for developers",
              "url": "https://front10.com",
            },
            Object {
              "name": "Another one",
              "url": "https://front10.com",
            },
            Object {
              "name": "Last time",
              "url": "https://front10.com",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <footer
      sections="[object Object]"
    />,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "sections": Array [
        Object {
          "name": "Features",
          "sections": Array [
            Object {
              "name": "Cool stuff",
              "url": "https://front10.com",
            },
            Object {
              "name": "Random feature",
              "url": "https://front10.com",
            },
            Object {
              "name": "Team feature",
              "url": "https://front10.com",
            },
            Object {
              "name": "Stuff for developers",
              "url": "https://front10.com",
            },
            Object {
              "name": "Another one",
              "url": "https://front10.com",
            },
            Object {
              "name": "Last time",
              "url": "https://front10.com",
            },
          ],
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <footer
        sections="[object Object]"
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "sections": Array [
          Object {
            "name": "Features",
            "sections": Array [
              Object {
                "name": "Cool stuff",
                "url": "https://front10.com",
              },
              Object {
                "name": "Random feature",
                "url": "https://front10.com",
              },
              Object {
                "name": "Team feature",
                "url": "https://front10.com",
              },
              Object {
                "name": "Stuff for developers",
                "url": "https://front10.com",
              },
              Object {
                "name": "Another one",
                "url": "https://front10.com",
              },
              Object {
                "name": "Last time",
                "url": "https://front10.com",
              },
            ],
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
